include/seastar/net/api.hh

```diff
@@ -129,6 +129,22 @@ public:
     void close();
 };
 
+class input_buffer_factory {
+public:
+  using buffer_t = temporary_buffer<char>;
+
+  virtual ~input_buffer_factory() = default;
+  /// Provide a rx buffer. Implementation is responsible for determining its size
+  /// and memory. This is useful when a network stack implementation does not put
+  /// extra requirements on these factors. The POSIX stack is the example here.
+  /// \param allocator Memory allocator \c connected_socket implementation prefers.
+  /// Maybe nullptr.
+  virtual buffer_t create(compat::polymorphic_allocator<char>* allocator) = 0;
```

Comment:
I understand this is not the final implementation, just for the record:

For input_buffer_factory of v2 protocol, `return seastar::make_temporary_buffer<char>(allocator,
      get_current_msg_size() + FRAME_PLAIN_EPILOGUE_SIZE + FRAME_PREAMBLE_SIZE + 41)` would be too simple, because it is the 3rd segment (DATA) which needs to be page-aligned.

Reply:
Yes, you're entirely correct. We would need to allocate a bigger buffer and `::trim_front` it accordingly.